File: corporate-reporting/src/types/KeyWorkArea.ts
import { ReportingEntity, IReportingEntity } from "./ReportingEntity";
import { IDirectorate } from "./Directorate";
import { IUser } from "./User";
import { IMilestone } from "./Milestone";
import { IKeyWorkAreaUpdate } from "./KeyWorkAreaUpdate";

export interface IKeyWorkArea extends IReportingEntity {
    DirectorateID: number;
    LeadUserID: number;
    RagOptionID: number;
    Directorate?: IDirectorate;
    KeyWorkAreaUpdates?: IKeyWorkAreaUpdate[];
    LeadUser?: IUser;
    Milestones?: IMilestone[];
}

export class KeyWorkArea extends ReportingEntity implements IKeyWorkArea {
    public DirectorateID = null;
    public LeadUserID = null;
    public RagOptionID = null;
}

File: ORB.Core/Models/KeyWorkArea.cs
using System;
using System.Collections.Generic;

namespace ORB.Core.Models
{
    public partial class KeyWorkArea : ReportingSubEntity
    {
        public KeyWorkArea()
        {
            KeyWorkAreaUpdates = new HashSet<KeyWorkAreaUpdate>();
            Milestones = new HashSet<Milestone>();
        }

        public int DirectorateID { get; set; }
        public int? RagOptionID { get; set; }

        public virtual Directorate Directorate { get; set; }
        public virtual RagOption RagOption { get; set; }
        public virtual ICollection<KeyWorkAreaUpdate> KeyWorkAreaUpdates { get; set; }
        public virtual ICollection<Milestone> Milestones { get; set; }
    }
}
